### DISCUSSION TITLE
link disabled in settings.json

### DISCUSSION THREAD

**Discussion by RoyChenCAD** [Yang-Xijie/yang-xijie.github.io#21] [Cpp in VSCode on macOS]
问题描述 
"cpp": "g++ $fullFileName stock01.cpp -o $dir\"$fileNameWithoutExt\"\".out\" -W -Wall -O2 -std=c++17 && $dir\"$fileNameWithoutExt\"\".out\"",

使用该命令行在settings.json中link stock01.cpp文件，但是运行usestock01.cpp文件仍然出现link 错误。这是文件代码
<img width="176" alt="Screenshot 2024-03-04 at 17 16 30" src="https://github.com/Yang-Xijie/yang-xijie.github.io/assets/153983553/bb72c226-0d74-4e0f-ad1a-c63d846119e7">

这是TERMINAL报错
a111@111deMacBook-Pro Cpp_Primer_Plus_6_Edition % g++ "/Users/a111/Desktop/Cpp_Primer_Plus
_6_Edition/10codes/usestock01.cpp" -o "/Users/a111/Desktop/Cpp_Primer_Plus_6_Edition/10cod
es/""usestock01"".out" -W -Wall -O2 -std=c++17 && "/Users/a111/Desktop/Cpp_Primer_Plus_6_E
dition/10codes/""usestock01"".out"
ld: Undefined symbols:
  Stock::buy(long, double), referenced from:
      _main in usestock01-b70e1a.o
  Stock::sell(long, double), referenced from:
      _main in usestock01-b70e1a.o
  Stock::show(), referenced from:
      _main in usestock01-b70e1a.o
      _main in usestock01-b70e1a.o
      _main in usestock01-b70e1a.o
  Stock::acquire(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, long, double), referenced from:
      _main in usestock01-b70e1a.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)

是否是因为执行了全局的.vscode settings.json的文件，而非该/10codes/.vscode setting.json的文件，请问我如何调试

**Comment by Yang-Xijie** [comment]
你最后说的原因是对的，因为你在 ./10codes/.vscode/settings.json 中添加的 `stock01.cpp` 并没有在命令行命令中出现。
解决方法，一个是，你可以去改 ./.vscode/settings.json 中的 `"cpp":` 后面的部分，添加 `./10codes/stock01.cpp`（注意这里使用相对路径）。另一个是，你可以直接把 10codes 这个文件夹作为工作区打开，也就是把这个文件夹拖到程序坞的 vscode 图标上打开。

**Comment by RoyChenCAD** [comment]
原来如此，难怪我在替换并配置了新的文件夹以后出现该问题，没有注意到workspace所对应的默认路径范围（当然之前也没有用过编译器的link功能就是了）。vscode永远在以一种意想不到的方式串联我这几个月来学到的知识。

感谢up主的回复与解答！！
